test/sqllogictest/arrays.slt

```diff
@@ -211,3 +211,27 @@ query T
 SELECT ARRAY['a', 'b', 'c'][3.5]
 ----
 NULL
+
+# Array equality
+
+query B
+SELECT ARRAY[1,2,3] = ARRAY[1,2,3]
+----
+true
+
+query B
+SELECT ARRAY[1,2,4] = ARRAY[1,2,3]
+----
+false
+
+query B
+SELECT ARRAY[1,2,4] = NULL
+----
+NULL
+
+# This behavior is surprising (one might expect that the result would be
```

Comment:
It might be worth adding a test that tries to compare arrays of two different types:
```
materialize=> SELECT ARRAY[1,2,3] = ARRAY['1','2','3'];
ERROR:  no overload for integer[] = text[]: arguments cannot be implicitly cast to any implementation's parameters; try providing explicit casts
```

Reply:
Done. Thanks!